File: scala-solution/AttackDrone.scala
import cwinter.codecraft.core.api._
import scala.util.Random

class AttackDrone extends DroneController {
  override def onTick(): Unit = {
    if (Random.nextInt(30) == 0) {
      val newDirection = 2 * math.Pi * Random.nextDouble()
      moveInDirection(newDirection)
    }
    if (weaponsCooldown <= 0) {
      for (drone <- dronesInSight) {
        if (drone.isEnemy && isInMissileRange(drone)) {
          shootMissiles(drone)
        }
      }
    }
  }
}

File: java-solution/AttackDrone.java
import cwinter.codecraft.core.api.*;
import java.util.Random;

class AttackDrone extends JDroneController {
  static Random rng = new Random();


  @Override public void onTick() {
    if (rng.nextInt(30) == 0) {
      Double newDirection = 2 * Math.PI * rng.nextDouble();
      moveInDirection(newDirection);
    }
    if (weaponsCooldown() <= 0) {
      for (Drone drone : dronesInSight()) {
        if (drone.isEnemy() && isInMissileRange(drone)) {
          shootMissiles(drone);
        }
      }
    }
  }
}
